File: advent_of_code/2023/day11/part1.cpp
#include <iostream>

using namespace std;

string map[135];
string map2[135];

int main(){
    for(int i =0; i < 135; i++){
        cin >> map[i];
    }
    for(int i =0; i < 135; i++){
        cin >> map2[i];
    }
    int maxi = 0;
    long long int res= 0;
    for(int i = 0; i < 135; i++){
        for(int k = 0; k < 130; k++){
            if(int(map[i][k]) > int('!')){
                for(int u = 0; u < 135; u++){
                    for(int l = 0; l < 130; l++){
                        if(int(map[u][l]) > int('!')){
                            res += abs(u-i) + abs(l-k) + (abs(int(map[i][k])-int(map[u][l])) + abs(int(map2[i][k])-int(map2[u][l])) )*1e6;
                            maxi = max(maxi, abs(int(map[i][k])-int(map[u][l])));
                        }
                    }
                }
            }
        }
    }
    cout << maxi << endl;
    cout << res/2;
    return 0;
}

File: advent_of_code/2024/day02/part1.rs
use std::cmp;
use std::io;

fn main() {
    let mut n = String::new();
    io::stdin().read_line(&mut n).unwrap();
    let n: i32 = n.trim().parse().unwrap();

    let mut res = 0;

    for _i in 0..n {
        let mut line = String::new();
        io::stdin().read_line(&mut line).unwrap();
        let line = line.trim().split(" ");

        let mut arr: Vec<i32> = Vec::new();

        for part in line {
            let num: i32 = part.parse().unwrap();
            arr.push(num);
        }

        let mut rising = false;
        let mut falling = false;
        let mut diff = 0;

        for i in 0..arr.len() - 1 {
            if arr[i] <= arr[i + 1] {
                rising = true;
            }
            if arr[i] >= arr[i + 1] {
                falling = true;
            }
            diff = cmp::max((arr[i] - arr[i + 1]).abs(), diff);
        }

        // println!("{rising} {falling} {diff}");

        if !(rising && falling) && diff <= 3 {
            res += 1;
        }
    }

    println!("{res}");
}
